### DISCUSSION TITLE
Effects

### DISCUSSION THREAD

**Discussion by JustinWolf11** [ScanMountGoat/ssbh_editor#173] [Ideas]
Why not make it so that SSBH Editor can visualize effects with animations? I understand It may be buggy as it would probably render when idle as well, but it would make it easier to test any effects packs without crashing SSBU for the tests

**Comment by ScanMountGoat** [comment]
This would be a cool feature but extremely involved to implement. As I mentioned in a recent announcement, I don't have any plans for additional changes beyond bug fixes. I can provide guidance on code changes if someone else wants to implement this. There are a number of challenges that would need to be solved like parsing the effect files in Rust, figuring out how the relevant emitters work, creating a renderer with wgpu, and the final integration with SSBH Editor.
